library/Booster/JsonRpc.hs

```diff
@@ -155,14 +159,40 @@ respond stateVar =
                             unless (null newModule.symbols) $
                                 throwE . AddModuleError $
                                     "Module introduces new symbols: " <> listNames newModule.symbols
-                     in case runExcept (checkModule >> addToDefinitions newModule state.definitions) of
+
+                            when nameAsId $
+                                case (Map.lookup (getId $ newModule.name) state.definitions, Map.lookup moduleHash state.definitions) of
+                                    (Just{}, Nothing) ->
+                                        -- another module with the same name already exists
+                                        throwE . AddModuleError $ "Duplicate Module Name"
+                                    (Just nmMod, Just idMod)
+                                        | nmMod /= idMod ->
+                                            -- this module has previously been added and different
+                                            -- module with the same name also already exists
+                                            throwE . AddModuleError $ "Duplicate Module Name"
```

Comment:
To return `DuplicateModuleName` to the client, we should use `DuplicateModule` here.
```suggestion
                                        throwE $ DuplicateModule $ getId newModule.name
                                    (Just nmMod, Just idMod)
                                        | nmMod /= idMod ->
                                            -- this module has previously been added and different
                                            -- module with the same name also already exists
                                            throwE $ DuplicateModule $ getId newModule.name
```